### PR TITLE
Flesh out "sharing knowledge" competency

### PR DESCRIPTION
**Pull request by MatMoore**
This is a first draft - what do people think? What do we want to say about this?

Update: [Second attempt here](https://github.com/alphagov/gds-tech-learning-pathway/blob/0d357337f6ee86873e996e21bd07f6f2f0b97e3c/career-path/competencies/sharing-knowledge-with-others.md)

### CODE REVIEW

File: career-path/competencies/sharing-knowledge-with-others.md [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+- Volunteer to speak at a monthly Technospective
+- Write a blog post for one of the GDS blogs
+- Speak at a conference
+
+### Mid level
 
-- Through writing documentation and guides
-- Talks
-- Writing blogs on what you have been working on
+At mid level you could also consider mentoring another developer, or helping a junior developer with a 20% project.
```
Comment (Davidslv) [inline career-path/competencies/sharing-knowledge-with-others.md]: Also could be: give/share presentations, do workshops, contribute ideas for juniors 20% projects.

File: career-path/competencies/sharing-knowledge-with-others.md [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+You should be open to talking about what you're working on.
+
+### Junior level
+
+Some ways you can share knowledge are:
+
+- Demonstrate work at Show and Tell events and within your team
+- Volunteer to speak at a monthly Technospective
+- Write a blog post for one of the GDS blogs
+- Speak at a conference
```
Comment (Davidslv) [inline career-path/competencies/sharing-knowledge-with-others.md]: I think this is a "nice to do" but most of the juniors don't feel confident enough to go and speak at conferences on subjects that they don't feel they have "mastered"

Comment (MatMoore) [reply career-path/competencies/sharing-knowledge-with-others.md]: Yup that's fair. Was a bit unsure how to structure this as it doesn't feel closely linked to job title 🤔

**Comment by bevanloon** [comment]
What about simply by pairing or mobbing. I don't think we should underestimate how much you can learn working together since you get to see other dev's workflow, environment and editor/ide config,  general approaches to problem solving etc.

**Comment by Davidslv** [comment]
Looking better, thanks for addressing the comments.
Another question: shouldn't we be mentioning reviewing pull requests? It's also a form of sharing knowledge between developers

**Comment by MatMoore** [comment]
Definitely, I'll add something in when I have a free moment.

**Comment by MatMoore** [comment]
I've added in something about code reviews, and linked to the styleguide on the subject, which has good guidance.

File: career-path/competencies/sharing-knowledge-with-others.md [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
+You should also regularly review other developers' pull requests. It's a good idea to check other work-in-progress before picking up new stories, so that code review doesn't become a bottleneck for the team.
+[The GDS styleguide](https://github.com/alphagov/styleguides/blob/master/pull-requests.md#guidance-for-each-step) has guidance about how to do good code reviews well.
+
+More suggestions:
 
-- Through writing documentation and guides
-- Talks
-- Writing blogs on what you have been working on
+- volunteer to mentor another developer
+- help a junior developer with a 20% project, or contribute ideas for 20% projects
```
Comment (Davidslv) [inline career-path/competencies/sharing-knowledge-with-others.md]: Should we link `contribute ideas` to this survey? https://docs.google.com/a/digital.cabinet-office.gov.uk/forms/d/e/1FAIpQLSeG8FP4mkaM1XYoY4rghgP7cUCXzLWNB849voZLVDGaWOxLQg/viewform

**Pull request merged**